components/board/genericlinux/data.go

```diff
@@ -129,76 +128,74 @@ func readIntFile(filePath string) (int, error) {
 	return int(resultInt64), err
 }
 
-func getGpioChipDefs(pinDefs []PinDefinition) (map[string]gpioChipData, error) {
-	gpioChipsInfo := map[string]gpioChipData{}
-	sysfsPrefixes := []string{"/sys/devices/", "/sys/devices/platform/", "/sys/devices/platform/bus@100000/"}
+func getGpioChipDefs(pinDefs []PinDefinition) (map[int]gpioChipData, error) {
+	gpioChipsInfo := map[int]gpioChipData{}
 
-	// Get a set of all the chip names with duplicates removed. Go doesn't have native set objects,
-	// so we use a map whose values are ignored.
-	gpioChipNames := make(map[string]struct{}, len(pinDefs))
-	for _, pinDef := range pinDefs {
-		if pinDef.GPIOChipSysFSDir == "" {
-			continue
-		}
-		gpioChipNames[pinDef.GPIOChipSysFSDir] = struct{}{}
-	}
+	allDevices := gpio.ChipDevices()
+	gpioChipNgpios := make(map[int]string, len(allDevices)) // maps chipNgpio -> string gpiochip#
+	for _, dev := range allDevices {
+		chip, _ := gpio.OpenChip(dev)
+		chipInfo, err := chip.Info()
 
-	for gpioChipName := range gpioChipNames {
-		var gpioChipDir string
-		for _, prefix := range sysfsPrefixes {
-			d := prefix + gpioChipName
-			fileInfo, err := os.Stat(d)
-			if err != nil {
-				continue
-			}
-			if fileInfo.IsDir() {
-				gpioChipDir = d
-				break
-			}
-		}
-		if gpioChipDir == "" {
-			return nil, errors.Errorf("cannot find GPIO chip %q", gpioChipName)
-		}
-		files, err := os.ReadDir(gpioChipDir)
 		if err != nil {
 			return nil, err
 		}
-		var chipFileName string
-		for _, file := range files {
-			if !strings.HasPrefix(file.Name(), "gpiochip") {
-				continue
-			}
-			chipFileName = file.Name()
-			break
-		}
 
-		gpioChipGPIODir := gpioChipDir + "/gpio"
-		files, err = os.ReadDir(gpioChipGPIODir)
-		if err != nil {
-			return nil, err
+		gpioChipNgpios[int(chipInfo.NumLines)] = chipInfo.Name
+	}
+
+	gpioConfigNgpios := make(map[int]struct{}, len(pinDefs))
+	for _, pinDef := range pinDefs {
+		for n := range pinDef.GPIOChipRelativeIDs {
+			gpioConfigNgpios[n] = struct{}{} // get a "set" of all ngpio numbers on the board
 		}
-		for _, file := range files {
-			if !strings.HasPrefix(file.Name(), "gpiochip") {
+	}
+
+	// TODO: remove this and base attribute after periph removed
+	sysPrefix := "/sys/class/gpio"
```

Comment:
This won't ever change, right? Make it a constant:
```go
const sysPrefix = "/sys/class/gpio"
```

and mild preference to rename sysfsPrefix, since this is not the system prefix.